core/src/main/scala/JwtClaim.scala

```diff
@@ -25,10 +25,14 @@ class JwtClaim(
   val issuedAt: Option[Long],
   val jwtId: Option[String]
 ) {
+
+  private final def simplifyToSingular[A](set: Set[A]): Option[A] = 
+    if (set.size == 1) set.headOption else None
+
   def toJson: String = JwtUtils.mergeJson(JwtUtils.hashToJson(Seq(
     "iss" -> issuer,
     "sub" -> subject,
-    "aud" -> audience,
+    "aud" -> audience.flatMap(set => simplifyToSingular(set).orElse(Some(set))),
```

Comment:
Can we not do just that? `headOption` does not seem useful since you already know there is an item inside the set.

```scala
"aud" -> audience.flatMap(set => if (set.size == 1) set.head else set),
```

Reply:
Yes that looks good to me

Reply:
I tried this but it doesn't compile.

Reply:
Just needed to use `map` rather than `flatMap` to make it work.